caluma/caluma_form/migrations/0047_recalculate_calc_answers.py

```diff
@@ -0,0 +1,55 @@
+from django.db import migrations
+
+from caluma.caluma_form.jexl import QuestionJexl
+from caluma.caluma_form.structure import FieldSet
+
+
+def recalculate_answer(answer, AnswerModel):
+    """
+    Recalculate a single calc-answer.
+
+    This is copy/paste from `_update_or_create_calc_answer` in order to use the old
+    models.
+    """
+    root_doc = answer.document.family
+
+    struc = FieldSet(root_doc, root_doc.form)
+    field = struc.get_field(answer.question.slug)
+
+    # skip if question doesn't exist in this document structure
+    if field is None:  # pragma: no cover
+        return
+
+    jexl = QuestionJexl(
+        {"form": field.form, "document": field.document, "structure": field.parent()}
+    )
+
+    # Ignore errors because we might evaluate partially built forms. So we might be
+    # missing some answers or the expression might be invalid, in which case we return
+    # None
+    value = jexl.evaluate(field.question.calc_expression, raise_on_error=False)
+
+    AnswerModel.objects.update_or_create(
+        question=answer.question, document=field.document, defaults={"value": value}
+    )
```

Comment:
You already have the answer object, so why the `update_or_create()` call?

Since this only affects already-calculated answers, you could just write 

```suggestion
    answer.value = value
    answer.save()
```

Reply:
Done.